SYCL/ESIMD/api/functional/operators/operator_logical_not.cpp

```diff
@@ -0,0 +1,161 @@
+//==------- operator_logical_not.cpp  - DPC++ ESIMD on-device test ---------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu, level_zero
+// XREQUIRES: gpu
+// TODO gpu and level_zero in REQUIRES due to only this platforms supported yet.
+// The current "REQUIRES" should be replaced with "gpu" only as mentioned in
+// "XREQUIRES".
+// UNSUPPORTED: cuda, hip
+// RUN: %clangxx -fsycl %s -fsycl-device-code-split=per_kernel -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+//
+// Test for simd logical not operator.
+// The test creates source simd instance with reference data and invokes logical
+// not operator.
+// The test verifies that data from simd is not corrupted after calling logical
+// not operator, that logical not operator return type is as expected and
+// logical not operator result values is correct.
+
+#include "../shared_element.hpp"
+#include "common.hpp"
+
+using namespace sycl::ext::intel::experimental::esimd;
+using namespace esimd_test::api::functional;
+
+// Descriptor class for the case of calling logical not operator.
+struct logical_not_operator {
```

Comment:
Would you mind to comment why do we need a separate descriptor class in this specific case?
Maybe some dependency, or architecture considerations, or some TODO for the future.

Without it I have a feeling that we could remove this class and move it contents into the `run_verification`

Reply:
I suggest do not do this change because we can add some other cases to this test by addition new classes like `logical_not_operator `, but if your comment will be applied we couldn't do it